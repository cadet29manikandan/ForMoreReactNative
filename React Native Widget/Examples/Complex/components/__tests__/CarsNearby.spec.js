import 'react-native';
import React from 'react';
import CarsNearby from '../CarsNearby';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

/* eslint-disable react/prefer-stateless-function */
jest.mock('../withDate', () => {
  const React = require('react');
  return ComposedComponent =>
    class extends React.Component {
      render() {
        return <ComposedComponent {...this.props} />;
      }
    };
});
/* eslint-disable react/no-multi-comp */
jest.mock('../withLocation', () => {
  const React = require('react');
  return ComposedComponent =>
    class extends React.Component {
      render() {
        return <ComposedComponent {...this.props} />;
      }
    };
});

jest.mock('../../lib/fetchBlob', () => ({
  dirs: { cacheDir: 'cache/dir/' },
  fetchToFile: () => Promise.resolve('resolved'),
  loadFromFile: () => mockCarsNearbyBlob,
}));

test('renders correctly', async () => {
  const component = await renderer.create(
    <CarsNearby latitude={50} longitude={14} date={new Date(2017, 12, 1)} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const mockCarsNearbyBlob = `[
    {
        "id": 685,
        "model": {
            "name": "Octavia Combi 2.0 TDI",
            "manufacturer": {
                "name": "Škoda"
            }
        },
        "local_total_price": {
            "amount": "993.00",
            "currency": "CZK",
            "verbose": "993.00 Kč"
        }
    },
    {
        "id": 617,
        "model": {
            "name": "Cee'd 1.6 CRDi 90",
            "manufacturer": {
                "name": "Kia"
            }
        },
        "local_total_price": {
            "amount": "993.00",
            "currency": "CZK",
            "verbose": "993.00 Kč"
        }
    },
    {
        "id": 562,
        "model": {
            "name": "CLK Coupe 320 CDI",
            "manufacturer": {
                "name": "Mercedes"
            }
        },
        "local_total_price": {
            "amount": "999.00",
            "currency": "CZK",
            "verbose": "999.00 Kč"
        }
    },
    {
        "id": 544,
        "model": {
            "name": "330d Cabrio",
            "manufacturer": {
                "name": "BMW"
            }
        },
        "local_total_price": {
            "amount": "1536.00",
            "currency": "CZK",
            "verbose": "1,536.00 Kč"
        }
    }
]`;
