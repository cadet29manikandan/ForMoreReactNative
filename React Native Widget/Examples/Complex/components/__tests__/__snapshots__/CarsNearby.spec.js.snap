// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "darkgray",
        "flex": 1,
        "minHeight": 55,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 55,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#2d2d2d",
              "flex": 3,
              "fontSize": 16,
            }
          }
        >
          Škoda Octavia Combi 2.0 TDI
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2d2d2d",
                "flex": 3,
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "textAlign": "right",
              },
            ]
          }
        >
          993 Kč
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "darkgray",
        "flex": 1,
        "minHeight": 55,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 55,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#2d2d2d",
              "flex": 3,
              "fontSize": 16,
            }
          }
        >
          Kia Cee'd 1.6 CRDi 90
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2d2d2d",
                "flex": 3,
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "textAlign": "right",
              },
            ]
          }
        >
          993 Kč
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "darkgray",
        "flex": 1,
        "minHeight": 55,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 55,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#2d2d2d",
              "flex": 3,
              "fontSize": 16,
            }
          }
        >
          Mercedes CLK Coupe 320 CDI
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2d2d2d",
                "flex": 3,
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "textAlign": "right",
              },
            ]
          }
        >
          999 Kč
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "darkgray",
        "flex": 1,
        "minHeight": 55,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 55,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#2d2d2d",
              "flex": 3,
              "fontSize": 16,
            }
          }
        >
          BMW 330d Cabrio
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2d2d2d",
                "flex": 3,
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "textAlign": "right",
              },
            ]
          }
        >
          1 536 Kč
        </Text>
      </View>
    </View>
  </View>
</View>
`;
